src/main/java/com/bandwidth/sqs/queue/buffer/task/DeleteMessageTask.java

```diff
@@ -0,0 +1,46 @@
+package com.bandwidth.sqs.queue.buffer.task;
+
+import com.bandwidth.sqs.actions.DeleteMessageBatchAction;
+import com.bandwidth.sqs.queue.buffer.entry.DeleteMessageEntry;
+import com.bandwidth.sqs.queue.buffer.task_buffer.Task;
+import com.bandwidth.sqs.request_sender.SqsRequestSender;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+public class DeleteMessageTask implements Task<String, DeleteMessageEntry> {
+
+    private final SqsRequestSender requestSender;
+
+    public DeleteMessageTask(SqsRequestSender requestSender) {
+        this.requestSender = requestSender;
+    }
+
+    @Override
+    public void run(String queueUrl, List<DeleteMessageEntry> entries) {
+        Map<String, DeleteMessageEntry> entryMap = new HashMap<>();
+        IntStream.range(0, entries.size()).forEach((id) -> {
+            entryMap.put(Integer.toString(id), entries.get(id));
+        });
```

Comment:
this is duplicated in all tasks, please extract method (`default` method on `Task`?)